api/envoy/api/v2/cluster.proto

```diff
@@ -438,6 +438,11 @@ message Cluster {
     // If set to `true`, the cluster manager will drain all existing
     // connections to upstream hosts whenever hosts are added or removed from the cluster.
     bool close_connections_on_host_set_change = 6;
+
+    // If set to `true`, the cluster will use hostname instead of the resolved
+    // name to load balance hosts. Only valid for StrictDNS and LogicalDNS
+    // clusters.
+    bool use_hostname = 7;
```

Comment:
Can we flesh out this documentation more and clarify what this actually does and that it actually has to do with hashing? I would pretty clearly indicate that it only works for hashing load balancers and provide some examples. Perhaps we should also call this `use_hostname_for_consistent_hashing` or something like that? WDYT?